addons/mail/controllers/thread.py

```diff
@@ -8,6 +8,22 @@
 
 
 class ThreadController(http.Controller):
+    @http.route("/mail/thread/access_rights", methods=["POST"], type="json", auth="user")
+    def mail_thread_access_rights(self, thread_data):
+        res = []
+        for thread_model in thread_data:
+            threads = request.env[thread_model].with_context(active_test=False).search([("id", "in", thread_data[thread_model])])
+            access_rights = threads._get_mail_thread_access_rights()
+            for thread_id in thread_data[thread_model]:
+                res.append(
+                    {
+                        "id": thread_id,
+                        "model": thread_model,
+                        **access_rights.get((thread_model, thread_id), {"hasReadAccess": False, "hasWriteAccess": False}),
```

Comment:
'_get_mail_thread_access_rights' already returns values for those keys, no need to over compensate, do we ?

Reply:
It was been necessary as the `_get_mail_thread_access_rights` would not return access rights for threads which user doesn't have read access. But, i have updated the logic in updated code. So, removed this part.